bindsnet/learning/__init__.py

```diff
@@ -385,46 +398,58 @@ def __init__(self, connection: AbstractConnection, nu: Optional[Union[float, Seq
                 'This learning rule is not supported for this Connection type.'
             )
 
-    def _connection_update(self, **kwargs) -> None:
+        self.p_plus = torch.zeros(self.source.n)
+        self.p_minus = torch.zeros(self.target.n)
+        self.eligibility = torch.zeros(self.source.n, self.target.n)
+
+    def _connection_update(self, dt: float, **kwargs) -> None:
         # language=rst
         """
-        M-STDP learning rule for ``Connection`` subclass of ``AbstractConnection`` class.
+        MSTDP learning rule for ``Connection`` subclass of ``AbstractConnection`` class.
+
+        :param dt: Simulation timestep.
 
         Keyword arguments:
 
         :param float reward: Reward signal from reinforcement learning task.
         :param float a_plus: Learning rate (post-synaptic).
         :param float a_minus: Learning rate (pre-synaptic).
+        :param tc_plus: Time constant for pre-synaptic firing trace.
+        :param tc_minus: Time constant for post-synaptic firing trace.
         """
 
+        # Reshape pre- and post-synaptic spikes.
         source_s = self.source.s.view(-1).float()
-        source_x = self.source.x.view(-1)
         target_s = self.target.s.view(-1).float()
-        target_x = self.target.x.view(-1)
 
+        # TODO: check this if necessary
```

Comment:
I think it is, because `w` needs to be 2D since `self.eligibility` is 2D. But you could just reshape `self.eligibility` before using it on line 436, which might be cleaner.

Reply:
I checked, and `self.connection.w` already is 2D. This is confirmed by the fact that the `.view` is missing in the `MSTDPET` rule, but it still functions.

https://github.com/Hananel-Hazan/bindsnet/blob/646dd7de73a1658bb622755e8bb3ebc406a265cd/bindsnet/learning/__init__.py#L542